src/config/backends/config-txt.ts

```diff
@@ -231,11 +232,21 @@ export class ConfigTxt extends ConfigBackend {
 	}
 
 	public isSupportedConfig(configName: string): boolean {
-		return !ConfigTxt.forbiddenConfigKeys.includes(configName);
+		return (
+			!ConfigTxt.forbiddenConfigKeys.includes(configName) &&
+			// power_mode and fan_profile are managed by the power-fan backend, so
+			// need to be excluded here as the config var name prefix is the same.
+			!isPowerFanSupportedConfig(configName)
```

Comment:
This breaks the independence between backends, why use the same prefix `HOST_CONFIG` for these new configurations rather than use something different like `HOST_OS` which is the way we would distinguish configurations before?

Reply:
Why not add it to `forbiddenConfigKeys` instead?

Reply:
Not sure about `HOST_OS`, the only occurrence of that I can see is the `HOST_OS_VERSION` env var. I'll move to adding to `forbiddenConfigKeys` instead.

Reply:
With latest iteration https://github.com/balena-os/balena-supervisor/pull/2379/commits/52f3258ffeb7f4dea1d5cb1d3f97bc250bff2bf1, forbiddenConfigKeys is now separated into forbidden and unsupported keys:
```
// These keys are not config.txt keys and are managed by the power-fan backend.
private static UNSUPPORTED_KEYS = ['power_mode', 'fan_profile'];
// These keys are config.txt keys, but are not mutable by the Supervisor.
private static FORBIDDEN_KEYS = [
	'disable_commandline_tags',
	'cmdline',
	'kernel',
	'kernel_address',
	'kernel_old',
	'ramfsfile',
	'ramfsaddr',
	'initramfs',
	'device_tree_address',
	'init_emmc_clock',
	'avoid_safe_mode',
].concat(ConfigTxt.UNSUPPORTED_KEYS);
```

The concept of unsupported keys is used by `isBootConfigVar` to filter out non-config.txt keys, but keep in forbidden config.txt keys such as initiramfs. When `isBootConfigVar` is called by `envToBootConfig` in `bootConfigChangeRequired`, this will allow `bootConfigChangeRequired` to provide the proper feedback to the user that they're attempting to set a forbidden (but not unsupported) key.

This still isn't ideal IMO, I think `isSupportedConfig`/`isBootConfigVar` are a bit redundant in terms of functionality. The config module could be simplified in this regard, but I would save this for another PR.